### DISCUSSION TITLE
Can you use custom metrics with fit? i.e. SMAPE

### DISCUSSION THREAD

**Discussion by pst2154** [Nixtla/mlforecast#138] [Q&A] [answered]
Is there a way to use SMAPE like in lightgbm?

```
lgb_ver_1 = lgb.LGBMRegressor().fit(X_train, y_train, eval_metric = 
                                   lambda y_true, y_pred: [lightgbm_smape(y_true, y_pred)])
````

**Comment by jmoralez** [comment, marked as answer]
Hey. I think you can do something like [this](https://github.com/Nixtla/mlforecast/issues/91#issuecomment-1446682677), that gives you full control on the fit.

**Comment by pst2154** [comment]
ok, this one runs but I'm not sure if its doing it correctly as there is no print message after the `fit`. Why does the model not give any fit information btw?

```
models = {
    'forecast': lgb.LGBMRegressor(**lgb_params, objective='regression', metric = 'lightgbm_smape'),
}
```

**Reply by jmoralez** [reply]
What do you mean?

**Reply by pst2154** [reply]
Sorry, I think i just mean getting model metrics out after you run the fit command. There kist does not seem to be much information from the lightGBM model stored in the forecast object after you call the fit method. Not a big deal

**Reply by jmoralez** [reply]
Are you calling the `MLForecast.fit` method? With the example above you can train the model in any way you like, e.g. using LightGBM's train API with callbacks and such.

**Reply by pst2154** [reply]
Ya I think I had a brainless moment when I wrote the question
